src/sentry/workflow_engine/processors/action.py

```diff
@@ -1,9 +1,71 @@
+from datetime import timedelta
+
+from django.db.models import DurationField, ExpressionWrapper, F, IntegerField, Value
+from django.db.models.fields.json import KeyTextTransform
+from django.db.models.functions import Cast, Coalesce
+from django.utils import timezone
+
 from sentry.db.models.manager.base_query_set import BaseQuerySet
-from sentry.workflow_engine.models import Action, DataConditionGroup, Workflow
+from sentry.models.group import Group
+from sentry.workflow_engine.models import Action, ActionGroupStatus, DataConditionGroup, Workflow
 from sentry.workflow_engine.processors.data_condition_group import evaluate_condition_group
 from sentry.workflow_engine.types import WorkflowJob
 
 
+def filter_recently_fired_workflow_actions(
+    actions: BaseQuerySet[Action], group: Group
+) -> BaseQuerySet[Action]:
+    # TODO(cathy): only reinforce workflow frequency for certain issue types
+    now = timezone.now()
+
+    statuses = ActionGroupStatus.objects.filter(group=group, action__in=actions)
```

Comment:
debating whether to add select_related here, i think it would make the annotation more performant? but is it going across too many FKs and/or there won't be too many objects here?

`.select_related("action__dataconditiongroupaction__condition_group__workflowdataconditiongroup__workflow__config")`

Reply:
i think given that we only need the 1 field, and how the number of joins that it will be pretty even on the performance. in general, select_related is more performant unless the tables are _huge_ and you want multiple properties.